solver/llbsolver/bridge.go

```diff
@@ -98,58 +122,93 @@ func (b *llbBridge) loadResult(ctx context.Context, def *pb.Definition, cacheImp
 			return nil, err
 		}
 	}
-	var cms []solver.CacheManager
-	for _, im := range cacheImports {
-		cmID, err := cmKey(im)
+
+	inlineCacheRemotes := map[digest.Digest]*solver.Remote{}
+
+	// Earthly custom inline cache handling. Other cache import types are ignored.
+	for _, cacheImport := range cacheImports {
+		if cacheImport.Type != "registry" {
+			continue
+		}
+
+		if b.importProcessed(cacheImport) {
+			continue
+		}
+
+		err := inBuilderContext(ctx, b.builder, "earthly inline cache", "", func(ctx context.Context, g session.Group) error {
+			inlineCacheRemotes, err = registry.EarthlyInlineCacheRemotes(ctx, b.sm, w, b.registryHosts, g, cacheImport.Attrs)
+			if err != nil {
+				return err
+			}
+			return nil
+		})
 		if err != nil {
-			return nil, err
+			bklog.G(ctx).Warnf("failed to import inline cache: %v", err)
```

Comment:
did you intentionally mean to drop the return here?

Reply:
Yes, it's probably not necessary to fail the build on an error. But now that you mention it, users may want to know. What do you think?

Reply:
Updated to return the error and fail the build. It's likely useful to know when a network error has occurred.